src/services/OpenKlantRegistrationService/Listener/strategies/PartijPerRolStrategy.ts

```diff
@@ -0,0 +1,154 @@
+    return persoon;
+  }
+
+  private async handleNietNatuurlijkPersoon(rol: Rol) {
+    // Create the partij
+    const partijInput = OpenKlantMapper.persoonPartijFromRol(rol);
+    const persoon = await this.configuration.openKlantApi.registerPartij(partijInput);
+    console.debug('Persoon partij created (for kvk in partij per rol strategy)', persoon);
+    // Create the partij identificatie
+    const identificatieInput = this.createTemporaryPartijIdentificatie(persoon.uuid);
+    const identificatie = await this.configuration.openKlantApi.addPartijIdentificatie(identificatieInput);
+    console.debug('Persoon partij identificatie created (for kvk in partij per rol strategy)', identificatie);
+    // Add the digitale adressen (and select voorkeur)
+    await this.setDigitaleAdressenForPartijFromRol(persoon, rol);
+    console.debug('Digitale addressen created (for kvk in partij per rol strategy)');
+    return persoon;
+
+  }
+
+  private async updateRolWithParijUrl(partijUuid: string, rol: Rol) {
+    const partijUrl = this.configuration.openKlantApi.getEndpoint() + `/partijen/${partijUuid}`;
+    rol.betrokkene = partijUrl;
+    const rolUpdate = await this.configuration.zakenApi.updateRol(rol);
+    console.debug('Rol updated with betrokkene', rolUpdate);
+  }
+
+  private async setDigitaleAdressenForPartijFromRol(partij: OpenKlantPartijWithUuid, rol: Rol) {
+    // Attach digitale adressen to partij
+    const digitaleAdressenInput = OpenKlantMapper.digitaalAdressenFromRol(rol, partij.uuid);
+    const promises: Promise<OpenKlantDigitaalAdresWithUuid>[] = [];
+    digitaleAdressenInput.forEach(digitaalAdres => {
+      promises.push(this.configuration.openKlantApi.addDigitaalAdres(digitaalAdres));
+    });
+    const digitaleAdressen = await Promise.all(promises);
+    digitaleAdressen.forEach(adres => console.log('Digitaal adres created', adres));
+
+    // Store the first digitaal adres as the prefered
+    // TODO figure out what the primary should be of the returned adressen?
+    const telefoon = digitaleAdressen.find(adres => adres.soortDigitaalAdres == OpenKlantMapper.TELEFOONNUMMER);
+    const email = digitaleAdressen.find(adres => adres.soortDigitaalAdres == OpenKlantMapper.EMAIL);
+    const voorkeur = telefoon?.uuid ?? email?.uuid;
+    if (!voorkeur) {
+      throw Error('No telefoon or email adres registered.');
+    }
+    const partijUpdate = await this.configuration.openKlantApi.updatePartij({
+      uuid: partij.uuid,
+      soortPartij: partij.soortPartij,
+      voorkeursDigitaalAdres: {
+        uuid: voorkeur,
+      },
+    });
+    console.debug('Partij updates with voorkeur', partijUpdate);
+  }
+
+  private createTemporaryPartijIdentificatie(partijUuid: string) {
+    return {
+      identificeerdePartij: {
+        uuid: partijUuid,
+      },
+      partijIdentificator: {
+        codeObjecttype: 'TEMPORARY PARTIJ ID',
+        codeSoortObjectId: 'tempPartijId',
+        objectId: randomUUID(),
```

Comment:
Ik dacht dat je iets wilde genereren gebaseerd op bestaande info. Dit kan natuurlijk ook

Reply:
Dit is denk ik beter, we willen namelijk expliciet niet dat we deze partijen terug kunnen vinden. Ze zijn alleen gerelateerd aan een rol met het uuid (niet het uuid hier maar die van het hele partij object). Het is wel handig als we uiteindelijk alle partijen met een codeObjecttype van `TEMPORARY PARTIJ ID` in een keer kunnen verwijderen in een later stadium.